edc-extensions/cx-oauth2/src/main/java/org/eclipse/tractusx/edc/oauth2/CxParticipantExtension.java

```diff
@@ -0,0 +1,74 @@
+/*
+ *  Copyright (c) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.tractusx.edc.oauth2;
+
+import org.eclipse.edc.runtime.metamodel.annotation.Inject;
+import org.eclipse.edc.runtime.metamodel.annotation.Setting;
+import org.eclipse.edc.spi.agent.ParticipantAgentService;
+import org.eclipse.edc.spi.agent.ParticipantAgentServiceExtension;
+import org.eclipse.edc.spi.iam.ClaimToken;
+import org.eclipse.edc.spi.monitor.Monitor;
+import org.eclipse.edc.spi.system.ServiceExtension;
+import org.eclipse.edc.spi.system.ServiceExtensionContext;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import static org.eclipse.edc.spi.agent.ParticipantAgent.PARTICIPANT_IDENTITY;
+
+public class CxParticipantExtension implements ServiceExtension, ParticipantAgentServiceExtension {
+
+    public static final String REFERRING_CONNECTOR_CLAIM = "referringConnector";
+
+    private static final String DEFAULT_PARTICIPANT_ID_REGEX = "[^/]+(?=/$|$)";
+    private static final int DEFAULT_PARTICIPANT_ID_REGEX_GROUP = 0;
+
+    @Setting(value = "Data plane proxy API consumer port", defaultValue = CxParticipantExtension.PARTICIPANT_ID_REGEX)
+    private static final String PARTICIPANT_ID_REGEX = "tx.participant.id.regex";
+
+    @Setting(value = "Data plane proxy API consumer port", defaultValue = CxParticipantExtension.PARTICIPANT_ID_REGEX)
+    private static final String PARTICIPANT_ID_REGEX_GROUP = "tx.participant.id.regex";
```

Comment:
same property as `PARTICIPANT_ID_REGEX`. is that intentional?

Reply:
nope